src/components/staff-payment/StaffPayment.vue

```diff
@@ -233,46 +232,27 @@ export default class StaffPayment extends Vue {
         this.$refs.bcolForm.resetValidation()
         this.$refs.folioNumberInputRef.resetFolioNumberValidation()
         // update data
-        this.emitStaffPaymentData({ option: StaffPaymentOptions.NO_FEE, isPriority: false })
+        this.staffPaymentData.isPriority = false
         break
     }
   }
 
-  /** Watches for change to FAS form validity. */
-  @Watch('fasFormValid')
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  private onFasFormValid (val: boolean) {
-    // ignore initial condition
-    if (!this.isMounted) return
-    this.emitValid()
-  }
-
-  /** Watches for change to BCOL form validity. */
-  @Watch('bcolFormValid')
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  private onBcolFormValid (val: boolean) {
-    // ignore initial condition
-    if (!this.isMounted) return
-    this.emitValid()
-  }
-
   /** Watches for changes to Staff Payment Data prop. */
   @Watch('staffPaymentData', { deep: true, immediate: true })
   private async onStaffPaymentDataChanged (val: StaffPaymentIF): Promise<void> {
-    this.paymentOption = val.option
-    await this.$nextTick()
+    this.emitStaffPaymentData({ ...this.staffPaymentData })
     this.emitValid()
   }
 
   /** Emits an event to update the Staff Payment Data prop. */
   @Emit('update:staffPaymentData')
   private emitStaffPaymentData ({
-    option = this.staffPaymentData.option,
-    routingSlipNumber = this.staffPaymentData.routingSlipNumber || '',
-    bcolAccountNumber = this.staffPaymentData.bcolAccountNumber || '',
-    datNumber = this.staffPaymentData.datNumber || '',
-    folioNumber = this.staffPaymentData.folioNumber || '',
-    isPriority = this.staffPaymentData.isPriority || false
+    option = StaffPaymentOptions.NONE,
+    routingSlipNumber = '',
+    bcolAccountNumber = '',
+    datNumber = '',
+    folioNumber = '',
+    isPriority = false
```

Comment:
The defaults on the left hand side of the `||` sign are no longer needed, since the current value from `staffPaymentData` will always get passed in line 243. We only need to handle null values.